src/SBVRType.d.ts

```diff
@@ -0,0 +1,47 @@
+interface SBVRType<I,O> {
+  types: ConcreteTypes
+
+  validate(value: any, required: boolean, cb:Callback<I>): void
+
+  fetchProcessing?(data:I, cb:Callback<O>): void
+
+  nativeProperties?: NativeProperties
+
+  nativeFactTypes?: NativeFactTypes
+}
+
+type Callback<T> = (err?: any, data?: T) => void
+
+declare class ConcreteTypes {
+  postgres: DBTypeBuilder
+  mysql: DBTypeBuilder
+  websql: DBTypeBuilder
+  odata: ODataTypeBuilder
+}
+
+type DBTypeBuilder =
+    string
+  | ((necessity: string, index: string) => string)
+
+declare class ODataTypeBuilder {
+  name: string
+  complexType?: string
+}
+
+// It would be nice to be a bit smarter about these verb and term types. Unfortunately indexed types can only be string or number atm and this excludes evern type aliases or a sum of possible string literals
```

Comment:
Actually, you can use a list of string literals as index types (if I understand 'sum of possible literals' correctly), by using the `in` keyword and an optional modifier:
```typescript
type Verb = 'run' | 'jump' | 'stand';
type VerbIndexedThing = { [v in Verb]?: string; }

const passes: VerbIndexedThing = {
    'run': 'I like running',
    'jump': 'I like jumping',
};

const fails VerbIndexedThing = {
    'stand': 'I like standing',
    'falling': 'I like falling',
};
```
`passes` is valid as it includes one of the allowed verbs, but `fails` isn't.